File: Java/프로그래머스/3/43238. 입국심사/입국심사.java
import java.util.*;

class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        long min = 0;
        long max = times[times.length - 1] * (long) n;
        Arrays.sort(times);
        while (min <= max) {
            long mid = (min + max) / 2;
            long count = 0;
            for (int time : times) {
                count += mid / time;
            }

            if (count >= n) {
                max = mid - 1;
                answer = mid;
            } else {
                min = mid + 1;
            }
        }
        return answer;
    }
}

File: Python/프로그래머스/3/43238. 입국심사/입국심사.py
def solution(n, times):
    answer = 0
    minimum = 0
    maximum = times[len(times) - 1] * n
    times.sort()

    while minimum <= maximum:
        mid = (minimum + maximum) // 2
        cnt = 0

        for time in times:
            cnt += mid // time

        if cnt >= n:
            maximum = mid - 1
            answer = mid
        else:
            minimum = mid + 1

    return answer